lua/nvim-tree/core.lua

```diff
@@ -25,6 +27,49 @@ function M.init(foldername)
   log.profile_end(profile)
 end
 
+---@param path string
+function M.change_root(path)
+  if TreeExplorer == nil then
+    return
+  end
+  local root_parent_cwd = vim.fn.fnamemodify(utils.path_remove_trailing(TreeExplorer.absolute_path), ":h")
+  if root_parent_cwd == path then
+    local newTreeExplorer = explorer.Explorer.new(path)
+    if newTreeExplorer == nil then
+      return
+    end
+    for _, node in ipairs(newTreeExplorer.nodes) do
+      if node.absolute_path == TreeExplorer.absolute_path then
+        node.nodes = TreeExplorer.nodes
```

Comment:
These nodes are being destroyed when `TreeExplorer:destroy()` is executed.

Test case:
create.sh
```sh
#!/bin/sh

mkdir -p a/1/A
mkdir -p a/2/A
mkdir -p a/2/B
mkdir -p b/1/A
mkdir -p b/2/B

touch foo
touch a/1/foo
touch a/1/A/bar
touch a/2/A/baz
touch a/2/B/foo
touch b/1/A/bar
touch b/2/B/baz
```

Open from directory `a` and expand all `E`.

From another terminal or cmd mode `touch a/1/A/zero` and see it appear.

`-` to change directory up.

`rm a/1/A/zero` does not update the tree until manual refresh `R`.

This is because the watcher has been destroyed.

![20240728_152354](https://github.com/user-attachments/assets/0c244a18-dc71-4bd5-89c6-4ce30983dbd7)

Reply:
Oh, I overlooked that fact. Let me make some changes. Thank you for providing the test case.